### PR TITLE
[New Page]: How to write a query runner

### PR DESCRIPTION
**Pull request by susodapop**
## Description

This PR adds a new page to the dev guide that discusses how to add a new Query Runner to Redash. I started with [Arik's instructions ](https://discuss.redash.io/t/creating-a-new-query-runner-data-source-in-redash/347) from December 2016 and updated them with the latest information. Rather than Snowflake, the doc page now uses the recently added [Firebolt Query Runner](https://github.com/getredash/redash/pull/5689) as an example.

### CODE REVIEW

File: src/pages/kb/open-source/dev-guide/write-a-query-runner.md [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+from redash.query_runner import BaseQueryRunner, register
+
+class Firebolt(BaseQueryRunner):
+    def run_query(self, query, user):
+        pass
+```
+
+The only method that you must implement is the `run_query` method, which accepts a `query` parameter (string) and the `user` who invoked this query. The user is irrelevant for most query runners and can be ignored.
+
+## Configuration
```
Comment (arikfr) [inline src/pages/kb/open-source/dev-guide/write-a-query-runner.md:24]: It's worth showing an example for:
1. Adding a friendly name and description for fields.
2. File type fields.

We should at least call out it's possible.

**Review by arikfr — APPROVED** [review]
Looks great. One comment about the config block.

File: src/pages/kb/open-source/dev-guide/write-a-query-runner.md (hunk 1 above)
Comment (susodapop) [reply src/pages/kb/open-source/dev-guide/write-a-query-runner.md:25]: Great catch. Fixed in bae95c6

**Comment by susodapop** [comment]
Alrighty ready to merge this. Had to sort out some confusion where I had built this PR on top of two commits that never merged to `master`. So when I looked at the changed files there were more than I expected. I sorted that by merging #620 and then rebasing this PR onto master.

**Pull request merged**